### PR TITLE
Build out more CI/CD documentation

### PR DESCRIPTION
**Pull request by damccorm**
Trying to continue to add more information to help folks self-govern their repos. Part of #1459

### CODE REVIEW

File: contributor-docs/cicd.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+See https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1573 for an example of splitting out the Spanner tests.
+
+## Runners
+
+There are 2 types of runners that CI/CD runs are performed on: GitHub hosted and self-hosted. These are configured in the `runs-on` section of each workflow. For example, `runs-on: [self-hosted, perf]` means that the workflow is running on a self-hosted runner for perf tests.
+
+For more information on GitHub hosted runners, see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+
+Google self-hosted runners are provisioned using the scripts defined in `https://github.com/manitgupta/DataflowTemplates/tree/main/.github/scripts`.
```
Comment (Abacn) [inline contributor-docs/cicd.md:34]: it links to a fork manitgupta/DataflowTemplates. Is this intended? Can we add it to the main repo?

**Review by Abacn — COMMENTED** [review]
thanks, read through the doc added had one comment here. @Polber who are familiar with template repo infra may have inputs

File: contributor-docs/cicd.md (hunk 1 above)
Comment (damccorm) [reply contributor-docs/cicd.md]: Yeah, fixed - thanks

**Comment by damccorm** [comment]
@Polber could you please take a look?

**Pull request merged**